static/site.js

```diff
@@ -1,4 +1,6 @@
 (function () {
+  const loginLink = document.getElementsByClassName('js-login-link')[0];
```

Comment:
> Then you could use "js-open-login-popup" in multiple places (if you changed the js code to find and attach to all instances of that class).

Actually if I make this change then it'll fix the big you pointed out in your second point as well, so I'll go ahead and do that